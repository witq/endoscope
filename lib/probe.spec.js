const { Probe } = require("./probe.js");

describe("Probe", () => {
  it("should pass with a long but not timed out probe", () => {
    const probe = () => new Promise(resolve => setTimeout(resolve, 10));
    const okCb = jest.fn();
    const errCb = jest.fn();

    return new Probe(probe, { timeout: 20 })
      .run()
      .then(okCb)
      .catch(errCb)
      .then(() => {
        expect(okCb).toHaveBeenCalled();
        expect(errCb).not.toHaveBeenCalled();
      });
  });

  it("should fail with not timed out but unsuccessful probe", () => {
    const probe = () => Promise.reject("error");
    const okCb = jest.fn();
    const errCb = jest.fn();

    return new Probe(probe, { timeout: 10 })
      .run()
      .then(okCb)
      .catch(errCb)
      .then(() => {
        expect(okCb).not.toHaveBeenCalled();
        expect(errCb).toHaveBeenCalledWith("error");
      });
  });

  it("should fail with timed out successfull probe", () => {
    const probe = () => new Promise(resolve => setTimeout(resolve, 20));
    const okCb = jest.fn();
    const errCb = jest.fn();

    return new Probe(probe, { timeout: 10 })
      .run()
      .then(okCb)
      .catch(errCb)
      .then(() => {
        expect(okCb).not.toHaveBeenCalled();
        expect(errCb).toHaveBeenCalled();
      });
  });

  it("should fail with timed out unsuccessfull probe", () => {
    const probe = () =>
      new Promise((_, reject) => setTimeout(() => reject("blblbl"), 20));
    const okCb = jest.fn();
    const errCb = jest.fn();

    return new Probe(probe, { timeout: 10 })
      .run()
      .then(okCb)
      .catch(errCb)
      .then(() => {
        expect(okCb).not.toHaveBeenCalled();
        expect(errCb).toHaveBeenCalled();
        expect(errCb).not.toHaveBeenCalledWith("blblbl");
      });
  });
});
