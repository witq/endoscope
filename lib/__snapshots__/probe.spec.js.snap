// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probe should return default message 1`] = `"OK"`;
